src/main/java/com/thealgorithms/scheduling/SpeculativeExecutionScheduling.java

```diff
@@ -0,0 +1,69 @@
+package com.thealgorithms.scheduling;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * SpeculativeExecutionScheduling runs multiple copies of the same task in parallel,
+ * picking the first one that finishes successfully. It is used to mitigate the
+ * impact of stragglers (slow tasks).
+ *
+ * Use Case: Big data systems like Hadoop and Spark, where it helps improve job completion time.
+ *
+ * @author Hardvan
+ */
+public final class SpeculativeExecutionScheduling {
```

Comment:
Scheduling should assign start time to all tasks, but this algorithm doesn't do that

Reply:
Done! I added the `startTime` variable.